dataflow/src/main/java/org/checkerframework/dataflow/cfg/builder/CFGTranslationPhaseOne.java

```diff
@@ -2358,7 +2359,12 @@ public Node visitErroneous(ErroneousTree tree, Void p) {
 
     @Override
     public Node visitExpressionStatement(ExpressionStatementTree tree, Void p) {
-        return scan(tree.getExpression(), p);
+        ExpressionTree exprTree = tree.getExpression();
+        scan(exprTree, p);
+        if (!(exprTree instanceof AssignmentTree)) {
```

Comment:
Why not also for `AssignmentTree`? If needed, can you add a comment that explains the motivation?